checkbox-provider-ce-oem/bin/tcp_multi_connections.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env python3
+import socket
+import argparse
+import threading
+import logging
+import time
+from datetime import datetime, timedelta
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[
+        logging.StreamHandler(),
+    ],
+)
+
+
+class PortOutputer():
+    def __init__(self, port,
+                 message="",
+                 list_status={}):
+        self.port = port
+        self.message = message
+        self.list_status = list_status
+        self._status = None
+        self._total_period = None
+        self._avg_time_period = None
+        self._max_time_period = None
+        self._min_time_period = None
+
+    @property
+    def status(self):
+        if not self.list_status:
+            return "ERROR"
+        else:
+            for check in self.list_status.values():
+                if check['status'] is False:
+                    self.message = "Received payload incorrect!"
+                    return "FAIL"
+            self.message = "Received payload correct!"
+            return "PASS"
```

Comment:
It seems to me that we will loop the list_values when getting the status property. I would suggest we parsing the data in the beginning.

e.g.
```
    def initial_data(self):
        total_value = timedelta()
        for value in self.list_status.values():
            total_value += value.get('time')

        self._total_period = total_value
        self._avg_time_period = total_value / len(self.list_status)

    @property
    def total_period(self):
        if self._total_period is None and self.list_status:
            self.initial_data()

        return self._total_period

    @property
    def avg_time_period(self):
        if self._avg_time_period is None and self.list_status:
            self.initial_data()

        return self._avg_time_period
```